### PR TITLE
Remove BuildConfigurations

### PR DESCRIPTION
**Pull request by jason-simmons**
Artifacts from local engine builds will be found based on the --local-engine flag

### CODE REVIEW

**Comment by jason-simmons** [comment]
@devoncarew

File: packages/flutter_tools/lib/src/build_configuration.dart [hunk 1]
```diff
@@ -81,37 +79,3 @@ HostPlatform getCurrentHostPlatform() {
...
-TargetPlatform getCurrentHostPlatformAsTarget() {
-  if (Platform.isMacOS)
-    return TargetPlatform.darwin_x64;
-  if (Platform.isLinux)
-    return TargetPlatform.linux_x64;
-  printError('Unsupported host platform, defaulting to Linux');
-  return TargetPlatform.linux_x64;
-}
-
-class BuildConfiguration {
```
Comment (devoncarew) [inline packages/flutter_tools/lib/src/build_configuration.dart]: We should rename this file from `build_configuration.dart` to something else - `builds.dart`? `build_info.dart`?

File: packages/flutter_tools/lib/src/commands/build_aot.dart [hunk 2]
```diff
@@ -132,7 +132,7 @@ String buildAotSnapshot(
     '--no-sim-use-hardfp',
   ];
 
-  if (!(tools.engineRelease || buildMode == BuildMode.release)) {
+  if (buildMode != BuildMode.release) {
```
Comment (devoncarew) [inline packages/flutter_tools/lib/src/commands/build_aot.dart]: I'm confused here. We're passing in `--no-checked` for profile and debug builds?

File: packages/flutter_tools/lib/src/commands/test.dart [hunk 3]
```diff
@@ -102,26 +84,16 @@ class TestCommand extends FlutterCommand {
...
-        return exitCode;
-    }
-    if (!foundOne) {
-      printError('At least one of --engine-debug or --engine-release must be set, to specify the local build products to test.');
+    loader.shellPath = tools.getHostToolPath(HostTool.SkyShell);
+    if (!FileSystemEntity.isFileSync(loader.shellPath)) {
+        printError('Cannot find Flutter shell at ${loader.shellPath}');
       return 1;
     }
+    await _runTests(testArgs, testDir);
```
Comment (devoncarew) [inline packages/flutter_tools/lib/src/commands/test.dart]: Could just `return await _runTests(testArgs, testDir);` here.

File: packages/flutter_tools/lib/src/toolchain.dart [hunk 4]
```diff
@@ -104,15 +69,25 @@ class ToolConfiguration {
...
+      return path.join(engineBuildPath, 'sky_snapshot');
+    } else if (tool == HostTool.SkyShell) {
+      if (getCurrentHostPlatform() == HostPlatform.linux_x64) {
+        return path.join(engineBuildPath, 'sky_shell');
+      } else if (getCurrentHostPlatform() == HostPlatform.darwin_x64) {
+        return path.join(engineBuildPath, 'SkyShell.app', 'Contents', 'MacOS', 'SkyShell');
+      }
     }
+
+    throw new Exception('Unexpected host tool: $tool');
```
Comment (devoncarew) [inline packages/flutter_tools/lib/src/toolchain.dart]: You can just throw a string here as well.

**Comment by devoncarew** [comment]
It looks like the only two remaining flags are `--local-engine-src-path` and `--local-engine`.

I'm confused by an if statement in `build_aot.dart`. Assuming that's the way it's supposed to be, lgtm.

File: packages/flutter_tools/lib/src/build_configuration.dart (hunk 1 above)
Comment (jason-simmons) [inline packages/flutter_tools/lib/src/build_configuration.dart]: done

File: packages/flutter_tools/lib/src/commands/build_aot.dart (hunk 2 above)
Comment (jason-simmons) [inline packages/flutter_tools/lib/src/commands/build_aot.dart]: Release builds of the AOT tools do not recognize these two flags

File: packages/flutter_tools/lib/src/commands/test.dart (hunk 3 above)
Comment (jason-simmons) [inline packages/flutter_tools/lib/src/commands/test.dart]: done

File: packages/flutter_tools/lib/src/toolchain.dart (hunk 4 above)
Comment (jason-simmons) [inline packages/flutter_tools/lib/src/toolchain.dart]: done

**Comment by devoncarew** [comment]
lgtm

File: packages/flutter_tools/lib/src/runner/flutter_command_runner.dart [hunk 5]
```diff
@@ -87,82 +73,14 @@ class FlutterCommandRunner extends CommandRunner {
...
-            'Path to your iOS Simulator Debug out directory, if you are building Flutter locally.\n'
-            'This path is relative to --engine-src-path. Not normally required.',
-        defaultsTo: 'out/ios_sim_Debug/');
-    argParser.addOption('ios-sim-release-build-path',
-        hide: !verboseHelp,
-        help:
-            'Path to your iOS Simulator Release out directory, if you are building Flutter locally.\n'
-            'This path is relative to --engine-src-path. Not normally required.',
-        defaultsTo: 'out/ios_sim_Release/');
+            'This path is relative to --local-engine-src-path/out.');
```
Comment (Hixie) [inline packages/flutter_tools/lib/src/runner/flutter_command_runner.dart]: this simplification is great.

**Pull request merged**